python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -238,39 +281,21 @@ def put(self):
 
         massStorageConfig = getConfig('Output')['MassStorageFile']['uploadOptions']
 
-        mkdir_cmd = massStorageConfig['mkdir_cmd']
         cp_cmd = massStorageConfig['cp_cmd']
         ls_cmd = massStorageConfig['ls_cmd']
         massStoragePath = massStorageConfig['path']
 
-        # create the last directory (if not exist) from the config path
-        import os.path
-        pathToDirName = os.path.dirname(massStoragePath)
-        dirName = os.path.basename(massStoragePath)
-
-        (exitcode, mystdout, mystderr) = self.execSyscmdSubprocess('%s %s' % (ls_cmd, pathToDirName))
-        if exitcode != 0:
-            self.handleUploadFailure(mystderr)
+        try:
+            self._mkdir(massStoragePath)
```

Comment:
I think that `mkdir -p`-style creation makes more sense to the user. It makes typos more 'dangerous' as it will create more unneeded directories but I think it will help more with the user's expectations.